ui/src/app/settings/components/project-details/project-details.tsx

```diff
@@ -473,225 +435,560 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
+                                                    formApi={formApi}
+                                                    field={`spec.sourceRepos[${i}]`}
+                                                    component={AutocompleteField}
+                                                    componentProps={{items: repos.map(repo => repo.repo)}}
+                                                />
+                                                <i className='fa fa-times' onClick={() => formApi.setValue('spec.sourceRepos', removeEl(formApi.values.spec.sourceRepos, i))} />
+                                            </div>
+                                        </div>
+                                    ))}
+                                    <button
+                                        className='argo-button argo-button--short'
+                                        onClick={() => formApi.setValue('spec.sourceRepos', (formApi.values.spec.sourceRepos || []).concat('*'))}>
+                                        ADD SOURCE
+                                    </button>
+                                </React.Fragment>
+                            )}
+                        </DataLoader>
+                    )}
+                    items={[]}
+                />
 
-                <h4>Blacklisted cluster resources {helpTip('Cluster-scoped K8s API Groups and Kinds which are not permitted to be deployed')}</h4>
-                {((proj.spec.clusterResourceBlacklist || []).length > 0 && (
-                    <div className='argo-table-list'>
-                        <div className='argo-table-list__head'>
-                            <div className='row'>
-                                <div className='columns small-3'>GROUP</div>
-                                <div className='columns small-6'>KIND</div>
-                            </div>
-                        </div>
-                        {(proj.spec.clusterResourceBlacklist || []).map(res => (
-                            <div className='argo-table-list__row' key={`${res.group}/${res.kind}`}>
-                                <div className='row'>
-                                    <div className='columns small-3'>{res.group}</div>
-                                    <div className='columns small-6'>{res.kind}</div>
-                                </div>
-                            </div>
-                        ))}
-                    </div>
-                )) || (
-                    <div className='white-box'>
-                        <p>No cluster-scoped resources are not permitted to deploy</p>
-                    </div>
-                )}
+                <EditablePanel
+                    save={item => this.saveProject(item)}
+                    values={proj}
+                    title={<React.Fragment>DESTINATIONS {helpTip('Cluster and namespaces where applications are permitted to be deployed to')}</React.Fragment>}
+                    view={
```

Comment:
Add header please so user can understand which means of each column:

```htmp
                    <React.Fragment>
                         <div className='row white-box__details-row'>
                                <div className='columns small-4'>Server</div>
                                <div className='columns small-8'>Namespace</div>
                            </div>
```

![image](https://user-images.githubusercontent.com/426437/94086756-b7f42380-fdc0-11ea-9ca4-2483bca65f47.png)